sources/niUI/src/Canvas.cpp

```diff
@@ -502,40 +502,10 @@ struct sCanvasVGPathTesselatedRenderer : public cIUnknownImpl<iVGPathTesselatedR
         eVGWrapType wrapType;
         eVGImageFilter imageFilter;
         if (apPaint->GetType() == eVGPaintType_Gradient) {
-          tU32 nResX = 128, nResY = 128;
-          const iVGPaintGradient* pPaintGrad = niStaticCast(const iVGPaintGradient*,apPaint);
-          switch (pPaintGrad->GetGradientType()) {
-          default:
-          case eVGGradientType_Linear:
-            {
-              wrapType = eVGWrapType_Mirror;
-              imageFilter = eVGImageFilter_Bilinear;
-              ptrImage = pPaintGrad->GetGradientTable()->CreateImage(pPaintGrad->GetGradientType(),pPaintGrad->GetWrapType(),NULL,nResX,1,pPaintGrad->GetD1(),pPaintGrad->GetD2());
-              break;
-            }
-          case eVGGradientType_Radial:
-          case eVGGradientType_Conic:
-          case eVGGradientType_SqrtCross:
-          case eVGGradientType_Cross:
-          case eVGGradientType_Diamond:
-            {
-              wrapType = eVGWrapType_Clamp;
-              imageFilter = eVGImageFilter_Bilinear;
-              Ptr<iVGTransform> transform = CreateVGTransform();
-              vScale *= pTransform->GetScaling()*0.5f;
-              vTrans.x = tF32(nResX/2);
-              vTrans.y = tF32(nResY/2);
-              transform->Scaling(vScale);
-              transform->Translate(vTrans);
-              ptrImage = pPaintGrad->GetGradientTable()->
-                  CreateImage(pPaintGrad->GetGradientType(),pPaintGrad->GetWrapType(),
-                              transform,
-                              nResX,nResX,pPaintGrad->GetD1(),pPaintGrad->GetD2());
-              vTrans /= vScale;
-              //VecInverse(vScale,vScale);
-              break;
-            }
-          }
+          QPtr<iVGPaintGradient> pPaintGradient(apPaint);
```

Comment:
QPtr<> could be null so it needs a null check. In the previous code `(pPaintGrad->GetGradientType())` mostly take care of that, but even that was pretty sketchy.